tests/jballerina-unit-test/src/test/resources/test-src/expressions/access/field_access.bal

```diff
@@ -353,10 +353,171 @@ function testFieldAccessOnMapConstruct() returns boolean {
     return "Sanjiva" == name;
 }
 
+type FooRec1 record {
+    json j;
+};
+
+type FooRec2 record {
+    json j;
+    FooRec1 fooRec;
+};
+
+type FooRec3 record {
+    json j?;
+    FooRec1 fooRec?;
+};
+
+function testFieldAccessOnJsonTypedRecordFields() returns error? {
+    FooRec1 rec1 = {j: "1"};
```

Comment:
Shall we update some test cases with longer strings since they're all single characters and can be Char

Reply:
Having that kind of test is not related to the actual issue we are fixing. So, it seems no need to add test like that. WDYT?

Reply:
Let's keep as is then.